homeassistant/components/light/tplink.py

```diff
@@ -40,7 +40,9 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     from pyHS100 import SmartBulb
     host = config.get(CONF_HOST)
     name = config.get(CONF_NAME)
-    add_devices([TPLinkSmartBulb(SmartBulb(host), name)], True)
+    bulb = TPLinkSmartBulb(SmartBulb(host), name)
+    bulb.get_features()
```

Comment:
Why not keep this call inside the TPLinkSmartBulb class?

Reply:
In `async_update_ha_state` the call to `update` will be done before any entity properties are accessed. I propose to revert all changes except on line 45, which will make sure that `update` is called before the first state update, and that supported features should be up to date.

Reply:
Hmm, if that's the case, there's then no need for this PR at all actually and the problem itself was fixed by #15484, as the second parameter was `True` already.